src/codegen/yul/statements.rs

```diff
@@ -465,3 +478,67 @@ fn process_for_block(
     cfg.set_phis(end_block, set.clone());
     cfg.set_phis(cond_block, set);
 }
+
+/// Generate CFG code for a switch statement
+fn resolve_switch(
```

Comment:
You'd expect `resolve_` in sema, not codegen. How about calling this `fn switch()`?